### PR TITLE
feat: Analytics outliers: round to two decimal digits, instead of only one[DHIS2-16954]

### PR DESCRIPTION
**Pull request by d-bernat**
All decimal numbers are now rounded to two decimal places. The default rounding scale remains unchanged and is set to 10 decimals.

### CODE REVIEW

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java [hunk 1]
```diff
@@ -293,11 +290,11 @@ public void queryOutliertest7() throws JSONException {
         false);
 
     // Assert rows.
-    validateRow(response, 0, List.of("O05mAByOgAv", "OPV2 doses given", "98.5"));
-    validateRow(response, 1, List.of("vI2csg55S9C", "OPV3 doses given", "70.3"));
-    validateRow(response, 2, List.of("n6aMJNLdvep", "Penta3 doses given", "70.1"));
-    validateRow(response, 3, List.of("UOlfIjgN8X6", "Fully Immunized child", "59.8"));
-    validateRow(response, 4, List.of("I78gJm4KBo7", "Penta2 doses given", "59.0"));
+    validateRow(response, List.of("O05mAByOgAv", "OPV2 doses given", "98.48"));
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java:293]: Is there a reason why you removed the indexes? If we have the `orderBy=modifiedzscore`, we should assert that the order is correct. I would keep the indexes. If the `orderBy` is not working it should be fixed...

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java [hunk 2]
```diff
@@ -340,11 +337,11 @@ public void queryOutliertest8() throws JSONException {
     validateHeader(response, 2, "zscore", "Z-score", "NUMBER", "java.lang.Double", false, false);
 
     // Assert rows.
-    validateRow(response, 0, List.of("l6byfWFUGaP", "Yellow Fever doses given", "3.7"));
-    validateRow(response, 1, List.of("s46m5MS0hxu", "BCG doses given", "3.1"));
+    validateRow(response, List.of("l6byfWFUGaP", "Yellow Fever doses given", "3.66"));
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java:340]: Is there a reason why you removed the indexes? If we have the `orderBy=zscore`, we should assert that the order is correct. I would keep the indexes. If the `orderBy` is not working it should be fixed...

**Review by maikelarabori — COMMENTED** [review]
Two minor comments, @d-bernat. Otherwise looks good, thx

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java]: done

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java]: done

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java [hunk 3]
```diff
@@ -340,11 +340,11 @@ public void queryOutliertest8() throws JSONException {
     validateHeader(response, 2, "zscore", "Z-score", "NUMBER", "java.lang.Double", false, false);
 
     // Assert rows.
-    validateRow(response, 0, List.of("l6byfWFUGaP", "Yellow Fever doses given", "3.7"));
+    validateRow(response, 0, List.of("l6byfWFUGaP", "Yellow Fever doses given", "3.66"));
     validateRow(response, 1, List.of("s46m5MS0hxu", "BCG doses given", "3.1"));
     validateRow(
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java:345]: I think this line is still missing the index...

Comment (d-bernat) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/outlier/OutliersDetection2AutoTest.java:345]: done

**Review by maikelarabori — APPROVED** [review]

**Pull request merged**